src/osdc/Journaler.cc

```diff
@@ -568,7 +568,7 @@ uint64_t Journaler::append_entry(bufferlist& bl)
     ldout(cct, 10) << " flushing completed object(s) (su " << su << " wro "
 		   << write_obj << " flo " << flush_obj << ")" << dendl;
     _do_flush(write_buf.length() - write_off);
-    if (write_off) {
+    if (write_off && write_buf.length() <= wrote) {
```

Comment:
I *think* you're trying to make sure we aren't getting partway into the written data with this update? But the conditional is pretty obtuse about that; pease add a comment.

Also, does this actually solve the problem? Seems like when adding entry N+1, we can split the commit of entry N. It'd be a lot more obvious to just look at which object we're actually flushing and move the next_safe_pointer to the end of that.